chore: added blog article: 'ASTs and TypeScript-ESLint' (#6039)
typescript-eslint/typescript-eslint: packages/website/blog/2022-12-01-asts-and-typescript-eslint.md

> In other words, an AST is a description of your code's syntax.

### An Example AST

Take this single line of code:

```js
1 + 2;
```

ESLint's AST format, **[ESTree]**, would describe that line of code as an object like:

```json
{
  "type": "ExpressionStatement",
  "expression": {
    "type": "BinaryExpression",
    "left": {
      "type": "Literal",
      "value": 1,
      "raw": "1"
    },
    "operator": "+",
    "right": {
      "type": "Literal",
      "value": 2,
      "raw": "2"
    }
  }
}
```

Each piece of code described within an AST description is referred to as a **node**, or AST node.
Each node is given a **node type** indicating the type of code syntax it represents
That code snippet includes four nodes of the following types:

- _ExpressionStatement_: `1 + 2;`
- _BinaryExpression_: `1 + 2`
- _Literal_: `1`
- _Literal_: `2`

That ESTree object representation of the code is what static analysis tools such as [ESLint](https://eslint.org) and [Prettier](https://prettier.io) work with.

## AST Formats

> ESTree is more broadly used than just for ESLint -- it is the de facto community standard.

Reviewer: ```suggestion
ESTree is more broadly used than just for ESLint -- it is a popular community standard.
```

it's not really a de-facto standard; there are a lot of different ASTs that are used (eg typescript's own AST). But it is probably the most popular
